cluster/sbin/collect-debug-artifacts.sh

```diff
@@ -186,18 +225,51 @@ function collect_on_remote {
         # also check for the pid line and get the pid
         proc_id=`sed -n 's/.*Process ID: \([0-9]\+\)$/\1/p' ${l}`
         if [ "${verbose}" = "1" ]; then
-          echo "Adding file ${l} to the array"
+          echo "Adding latest copyright header file ${l} to the array"
         fi
         files+=($l)
         last_restart_log="$l"
+      else
+        if [ "${verbose}" = "1" ]; then
+          echo "Adding file ${l} to the array"
+        fi
+        files+=($l)
       fi
     done
 
     # get all the gfs files as well
-    for l in $( ls *.gfs 2> /dev/null )
+    for l in $( ls *.gfs* 2> /dev/null )
     do
       files+=($l)
     done
+  else 
+    if [ "${verbose}" = "1" ]; then
+      echo "collecting files based on modified time"
+    fi
+    files=()
+    for l in $( ls -t *.log* 2>/dev/null )
+    do
+      file_mod_epoch=`date +%s --date "$(ls  -l --time-style=long-iso $l | cut -d' ' -f6-7)"`
+      if [ "${file_mod_epoch}" -ge "${start_epoch}" -a "${file_mod_epoch}" -le "${end_epoch}" ]; then
+        if [ "${verbose}" = "1" ]; then
+          echo "${l} MOD TIME = ${file_mod_epoch}"
+          echo "Adding file ${l} to the array"
+        fi
+        files+=($l)
+      fi
+    done
+
+    for l in $( ls -t *.gfs* 2>/dev/null )
+    do
+      file_mod_epoch=`date +%s --date "$(ls  -l --time-style=long-iso $l | cut -d' ' -f6-7)"`
+      if [ "${file_mod_epoch}" -ge "${start_epoch}" -a "${file_mod_epoch}" -le "${end_epoch}" ]; then
```

Comment:
In other words if we keep `prev_file_mod_epoch`, then the condition will be:
```sh
if [ "${file_mod_epoch}" -ge "${start_epoch}" -a "${prev_file_mod_epoch}" -le "${end_epoch}" ]; then
```

Or in other words if there is any overlap of `[prev_file_mod_epoch - file_mod_epoch]` and `[start - end]` intervals. Don't need the next file time since this file's time is its last modification time.